build.sbt

```diff
@@ -37,7 +37,8 @@ lazy val commonSettings = Seq(
     compilerPlugin("org.scalamacros" %% "paradise" % "2.1.0-M5" cross CrossVersion.full),
     compilerPlugin("org.spire-math" %% "kind-projector" % "0.6.3")
   ),
-  parallelExecution in Test := false
+  parallelExecution in Test := false,
+  scalacOptions in (Compile, doc) := (scalacOptions in (Compile, doc)).value.filter(_ != "-Xfatal-warnings")
```

Comment:
@dwijnand Nice!—I had no idea that would work.

My approach to dealing with SBT is 95% cargo-cultery, and there's a similar `filter` a little further down in the file—I'll update both in this PR when I'm back at my desktop.